share/picongpu/tests/Pusher/lib/python/test/Pusher/MainTest.py

```diff
@@ -0,0 +1,325 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2023 PIConGPU contributors
+Authors: Hannes Wolf
+License: GPLv3+
+
+Test script to determine if a pusher works correctly. This is done by the
+comparison of the radii of a particle at constatnt speed in a homogenious
+magnetic field. If the relative change between two periods is greater than 
+epsilon = 1e-5, the test fails. Also the absolute phasechange during one turn
+is regarded. It should be smaller than 0.2 rad.
+"""
+
+import openpmd_api as io
+import numpy as np
+import sys
+
+
+def get_params(path):
+    """creates a series of the data found at path.
+    path should have the form data_%T to provide access to all iterations.
+
+    returns:
+        series
+        parameter list... list of parameters of the simulation (units etc.)
+        """
+
+    # load series
+    series = io.Series(path, io.Access.read_only)
+
+    if len(series.iterations) == 1:
+        print("There is just 1 iteration in the series")
+        print("make sure, there are at least two")
+        longer_series = False
+
+    else:
+        longer_series = True
+
+    # read order assignment function
+    i = series.iterations[0]
+
+    # read parameters of the simulation
+    cell_depth = i.get_attribute("cell_depth")
+    cell_height = i.get_attribute("cell_height")
+    cell_width = i.get_attribute("cell_width")
+    dt = i.get_attribute("dt")
+    unit_time = i.get_attribute("unit_time")
+    unit_length = i.get_attribute("unit_length")
+    unit_charge = i.get_attribute("unit_charge")
+    unit_bfield = i.get_attribute("unit_bfield")
+    unit_speed =  i.get_attribute("unit_speed")
+    unit_mass = i.get_attribute("unit_mass")
+
+    return longer_series, series, [
+        cell_depth, cell_height, cell_width, dt, unit_time, unit_charge,
+        unit_length, unit_mass, unit_speed, unit_bfield
+        ]
+
+def read_series(series):
+    """reads the position_offset (grid_knot) and position
+    (float in the intervall [0,1), relativ position in the cell) and the
+    momentum of the particle in the x-y-plane.
+
+    returns:
+        x/y_poss... array of all relative in-Cell positions
```

Comment:
what is pos**s**? What does the second "s" stand for?

Reply:
It is the array of all the **pos**ition**s**. So it is the plural of the abbreviation "pos" (for position).